compile.js

```diff
@@ -24,17 +30,35 @@ function writeContext(ctx, options) {
 }
 
 function writeMessage(ctx, options) {
+    const name = ctx._name;
     const fields = ctx._proto.fields;
 
     let code = '\n';
 
+    if (options.jsDoc) {
+        code += compileType(ctx, name, fields);
+    }
+
     if (!options.noRead) {
-        const readName = `read${ctx._name}`;
+        const readName = `read${name}`;
+        if (options.jsDoc) {
+            code += ['\n/**', ' * @param {Pbf} pbf', ' * @param {number} [end]', ` * @returns {${name}}`, ' */'].join('\n').concat('\n');
```

Comment:
Nit: it's probably nicer to use a multiline template literal here and similar places, like in a few other spots of `compile.js`.

Reply:
Good point, updated 👍